src/vs/workbench/parts/debug/electron-browser/debugEditorContribution.ts

```diff
@@ -296,6 +305,47 @@ export class DebugEditorContribution implements IDebugEditorContribution {
 		}
 	}
 
+	// exception widget
+	public toggleExceptionWidget(): void {
+		// Toggles exception widget based on the state of the current editor model and debug stack frame
+		const model = this.editor.getModel();
+		const focusedSf = this.debugService.getViewModel().focusedStackFrame;
+		const callStack = focusedSf ? focusedSf.thread.getCallStack() : null;
+		if (!model || !focusedSf || !callStack) {
+			this.closeExceptionWidget();
+			return;
+		}
+
+		// First call stack frame is the frame where exception has been thrown
+		const exceptionSf = callStack[0];
+		const sameUri = exceptionSf.source.uri.toString() === model.uri.toString();
+		if (this.exceptionWidget && !sameUri) {
+			this.hideExceptionWidget();
+		} else if (focusedSf && focusedSf.thread.stoppedDetails.reason === 'exception' && sameUri) {
+			this.showExceptionWidget(exceptionSf.lineNumber);
+		}
+	}
+
+	private showExceptionWidget(lineNumber: number): void {
+		if (this.exceptionWidget) {
+			this.exceptionWidget.dispose();
+		}
+
+		this.exceptionWidget = this.instantiationService.createInstance(ExceptionWidget, this.editor, lineNumber);
+		this.exceptionWidget.show({ lineNumber, column: 1 }, 3);
+	}
+
+	private hideExceptionWidget() {
+		this.exceptionWidget.hide();
+	}
+
+	private closeExceptionWidget(): void {
```

Comment:
No need to make difference between hide and close. Always destroy and recreate the widget as needed